### DISCUSSION TITLE
Valetudo "Unable to reach vacuum."

### DISCUSSION THREAD

**Discussion by alistair23** [Hypfer/Valetudo#1843] [Q&A / Support]
I have a:
- Dreame L10 Pro
- Firmware version: 1138
- Valetudo: 2023.05.0

It has been running Valetudo fine for about a year.

Suddenly though I'm unable to interact with the robot via Valetudo. I can access the web interface on the device, but everything just spins forever.

If I SSH into the device I see things like this in the `valetudo.log`

```
[2023-06-23T04:24:36.555Z] [INFO] Got token from handshake: 6253756d325764496232556e34436c34
[2023-06-23T04:24:37.855Z] [WARN] Unable to reach vacuum. Giving up after 100 tries
[2023-06-23T04:24:37.857Z] [ERROR] Failed auto refresh: RetryWrapperSurrenderError [MiioTimeoutError]: re
quest timed out:{"method":"get_properties","params":[{"siid":4,"piid":2,"did":521303417},{"siid":4,"piid"
:3,"did":521303417}],"id":1252528870}
    at RetryWrapper.retryHandshakeHelper (/snapshot/Valetudo/backend/lib/miio/RetryWrapper.js:182:19)
    at doRetry (/snapshot/Valetudo/backend/lib/miio/RetryWrapper.js:134:22)
    at /snapshot/Valetudo/backend/lib/miio/RetryWrapper.js:152:25
```

The robot still responds to button presses, so I can still manually run a cleaning cycle. It just seems like Valetudo and the firmware aren't communicating. Any ideas how to debug or fix this?

**Comment by Hypfer** [comment]
The issue you've reported is usually caused by the user joining the robot to a Wi-Fi network using some shell commands and/or editing config files. [As noted in the docs](https://valetudo.cloud/pages/general/getting-started.html#joining_wifi) this is not recommended as it will lead to issues such as this one.

The reason for that is that being connected to a Wi-Fi network and being provisioned are two different things. Here it is important to keep in mind that we're simply replacing the cloud. That means that the logic still assumes that the device would be bound to a specific MiHome account and only if it has that data would it start responding to commands and reporting data because only then the firmware would know who is allowed to access it etc.

Configuring Wi-Fi using Valetudo takes care of this correct provisioning process. Running any random shell commands to join a Wi-Fi network does not.

To fix this, simply reset Wi-Fi by pressing and holding the two outer buttons until the robot talks to you and then connect to the Wi-Fi AP and configure Wi-Fi [as documented in the docs](https://valetudo.cloud/pages/general/getting-started.html#joining_wifi)
